packages/trie/src/trie.ts

```diff
@@ -401,18 +401,16 @@ export class Trie {
    */
   async lookupNode(node: Uint8Array | Uint8Array[]): Promise<TrieNode | null> {
     if (isRawNode(node)) {
-      return decodeRawNode(node as Uint8Array[])
+      return decodeRawNode(node)
     }
-    let value = null
-    let foundNode = null
-    value = await this._db.get(node as Uint8Array)
-    if (value) {
-      foundNode = decodeNode(value)
-    } else {
+    const value = (await this._db.get(node)) ?? null
+
+    if (value === null) {
       // Dev note: this error message text is used for error checking in `checkRoot`, `verifyProof`, and `findPath`
       throw new Error('Missing node in DB')
     }
-    return foundNode
+
+    return decodeNode(value) ?? null
```

Comment:
Can `decodeNode` ever return undefined?  It's not typed for that.

Reply:
I wondered the same, and there seems to be some erroneous types in either this method or in instances where this method is used. I don't see how this could return undefined/null, but at the same time, in usage, we often check for the truthiness of the `lookupNode` return value, which leads me to think that it can sometimes return a nullish value. I'll investigate further and experiment with removing null completely (seems to me like it should either return a TrieNode or null)

Reply:
I removed `null` from the return value, let's see how the CI runs.